kde-apps/krdc/patches/krdc-24.02.2.patchset

```diff
@@ -0,0 +1,117 @@
+From 52535261201ddc351862e14f0c9116c1fe1c3dd6 Mon Sep 17 00:00:00 2001
+From: Sergei Reznikov <diver@gelios.net>
+Date: Fri, 24 Aug 2018 17:55:51 +0300
+Subject: Fix build on Haiku
+
+
+diff --git a/cmake/modules/FindLibSSH.cmake b/cmake/modules/FindLibSSH.cmake
+index c1fecfd..9dc6669 100644
+--- a/cmake/modules/FindLibSSH.cmake
++++ b/cmake/modules/FindLibSSH.cmake
+@@ -17,6 +17,8 @@ find_path(LIBSSH_INCLUDE_DIR
+     /usr/local/include
+     /opt/local/include
+     /sw/include
++	/system/develop/headers
++	/system/develop/headers/x86
+     ${CMAKE_INCLUDE_PATH}
+     ${CMAKE_INSTALL_PREFIX}/include
+ )
+@@ -30,6 +31,8 @@ find_library(SSH_LIBRARY
+     /usr/local/lib
+     /opt/local/lib
+     /sw/lib
++	/system/develop/headers
++	/system/develop/headers/x86
+     ${CMAKE_LIBRARY_PATH}
+     ${CMAKE_INSTALL_PREFIX}/lib
+ )
+diff --git a/vnc/CMakeLists.txt b/vnc/CMakeLists.txt
+index 20f74f0..6dd2933 100644
+--- a/vnc/CMakeLists.txt
++++ b/vnc/CMakeLists.txt
+@@ -31,6 +31,10 @@ target_link_libraries(krdc_vncplugin
+     krdccore
+ )
+ 
++if(HAIKU)
++    target_link_libraries(krdc_vncplugin network)
++endif ()
++
+ if(QT_MAJOR_VERSION STREQUAL "6")
+     target_link_libraries(krdc_vncplugin KF6::KCMUtils)
+ endif()
+diff --git a/vnc/vncclientthread.cpp b/vnc/vncclientthread.cpp
+index 5b5804c..cf96aed 100644
+--- a/vnc/vncclientthread.cpp
++++ b/vnc/vncclientthread.cpp
+@@ -668,7 +668,7 @@ void VncClientThread::clientSetKeepalive()
+     }
+ 
+     optval = m_keepalive.intervalSeconds;
+-    if (setsockopt(cl->sock, IPPROTO_TCP, TCP_KEEPIDLE, &optval, optlen) < 0) {
++/*    if (setsockopt(cl->sock, IPPROTO_TCP, TCP_KEEPIDLE, &optval, optlen) < 0) {
+         qCritical(KRDC) << "setsockopt(TCP_KEEPIDLE)" << strerror(errno);
+         return;
+     }
+@@ -684,7 +684,7 @@ void VncClientThread::clientSetKeepalive()
+         qCritical(KRDC) << "setsockopt(TCP_KEEPCNT)" << strerror(errno);
+         return;
+     }
+-    m_keepalive.set = true;
++*/    m_keepalive.set = true;
+     qCDebug(KRDC) << "TCP keepalive set";
+ }
+ 
+-- 
+2.43.2
+
+
+From d9af26051076119b1b02143d950b7847006a0627 Mon Sep 17 00:00:00 2001
+From: Schrijvers Luc <begasus@gmail.com>
+Date: Sat, 20 Apr 2024 12:37:21 +0200
+Subject: build fix
+
+
+diff --git a/rdp/rdpsession.cpp b/rdp/rdpsession.cpp
+index e4a3c28..e7c98c0 100644
+--- a/rdp/rdpsession.cpp
++++ b/rdp/rdpsession.cpp
+@@ -407,7 +407,9 @@ bool RdpSession::start()
+     }
+ 
+     m_thread = std::thread(std::bind(&RdpSession::run, this));
++#ifndef __HAIKU__
+     pthread_setname_np(m_thread.native_handle(), "rdp_session");
```

Comment:
Could this be implemented using [rename_thread](https://www.haiku-os.org/legacy-docs/bebook/TheKernelKit_ThreadsAndTeams.html#rename_thread) instead?

Reply:
I'm afraid my skills aren't up to that, maybe you can push a "proper" fix to this PR for it? :)

Reply:
It was just a suggestion, I'm not sure if this works either. I can probably have a look later, but probably not before the weekend.

Reply:
Something like?:
```
rename_thread(find_thread(NULL), buf) == B_OK;
```

Reply:
No, that would rename the current thread, not the new one. Instead of `find_thread(NULL)`, keep the `m_thread.native_handle()` (maybe you need a cast to `thread_id`). So just the function name is different.

Reply:
Thanks :) I'll leave it hanging for a while, has to wait for freerdp anyway.

Reply:
pthread_setname_np should be available in r1beta4, shouldn't it?

Reply:
seems defining _GNU_SOURCE before pthread.h is needed.

Reply:
OK, this works also (thanks @korli), I don't see a pthrread.h in there though:
```diff
--- a/rdp/rdpsession.cpp
+++ b/rdp/rdpsession.cpp
@@ -3,6 +3,8 @@
  * SPDX-License-Identifier: GPL-2.0-or-later
  */
 
+#define _GNU_SOURCE
+
 #include "rdpsession.h"
 
 #include <memory>
```

Reply:
Then adding -DCMAKE_CXX_FLAGS="-D_GNU_SOURCE" to the cmake invocation in the recipe should be enough.

Reply:
This one fixed, thanks!